.github/scripts/simple-smoke-test.sh

```diff
@@ -29,6 +30,18 @@ function stop_container() {
     docker stop "${container_name}"
 }
 
+function check_java_version() {
+    local expected_major_version=$1
+    echo "Checking Java version - expected ${expected_major_version}"
+    local actual_major_version
+    actual_major_version=$(docker run --rm "${image}" sh -c 'java -version 2>&1 | head -n 1 | awk -F "\"" "{print \$2}" | awk -F "." "{print \$1}"')
```

Comment:
nit - no need for `sh -c`?
i think default is `sh`

Reply:
> nit - no need for `sh -c`? i think default is `sh`

Were passing the entire `java -version 2>&1 | head -n 1 | awk -F "\"" "{print \$2}" | awk -F "." "{print \$1}"` command into the Docker image to execute.

If you run just java -version 2>&1` on the Docker image, and then pipe that into `head`, [you get a bizarre SIGPIPE failure](https://stackoverflow.com/a/19120674) - but only when running on the GitHub runner, not locally 

_ideally_ we shouldn't execute those commands in the container as were assuming it's got `head`, `awk` etc. But I wasted many hours on this.

Reply:
Understood. thanks